src/ApplicationInsights.Kubernetes/K8sInfoBackgroundService.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights.Kubernetes.Debugging;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.ApplicationInsights.Kubernetes;
+
+/// <summary>
+/// A bootstrap service to keep updating K8s environment.
+/// </summary>
+internal class K8sInfoBackgroundService : BackgroundService
+{
+    private readonly ApplicationInsightsKubernetesDiagnosticSource _logger = ApplicationInsightsKubernetesDiagnosticSource.Instance;
+    private readonly IServiceProvider _serviceProvider;
+    private readonly K8sEnvironmentHolder _k8SEnvironmentHolder;
+    private readonly AppInsightsForKubernetesOptions _options;
+
+    public K8sInfoBackgroundService(
+        IServiceProvider serviceProvider,
+        K8sEnvironmentHolder k8SEnvironmentHolder,
+        IOptions<AppInsightsForKubernetesOptions> options)
+    {
+        _serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+        _k8SEnvironmentHolder = k8SEnvironmentHolder ?? throw new ArgumentNullException(nameof(k8SEnvironmentHolder));
+        _options = options?.Value ?? throw new ArgumentNullException(nameof(options));
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        while (true)
+        {
+            try
+            {
+                _logger.LogDebug($"Starting {nameof(UpdateK8sEnvironmentAsync)}");
```

Comment:
If `stoppingToken` is never checked, how will this service shut down?

Reply:
Of course :-) I'll update the code.

Reply:
Thanks for pointing out.